### PR TITLE
new vignette: Benchmarking datatable

### LINKED ISSUE #2635: memory issue in devel when doing subset
```r
library(data.table)
packageVersion("data.table")

n = 1e6
K = 100
set.seed(1)
DT = data.table(
    char1 = sample(sprintf("id%010d",1:(n/K)), n, TRUE),
    char2 = sample(sprintf("id%03d",1:K), n, TRUE),
    char3 = sample(sprintf("id%1d",1:5), n, TRUE),
    int1 = sample(n/K, n, TRUE),
    int2 = sample(K, n, TRUE),
    int3 = sample(5, n, TRUE),
    double1 = sample(seq(0, 1, length.out=n/K), n, TRUE),
    double2 = sample(seq(0, 1, length.out=K), n, TRUE),
    double3 = sample(seq(0, 1, length.out=5), n, TRUE)
)

# query subset filters
subn = 0.001
res = DT[sample(n, n*subn, FALSE)]
cin1 = res[["char1"]]
iin1 = res[["int1"]]
din1 = res[["double1"]]
cin1 = unique(cin1)
iin1 = unique(iin1)
din1 = unique(din1)

# query
DT[char1%in%cin1 & int1%in%iin1 & double1%in%din1]
```
output on 1.10.4.3
```
[1] ‘1.10.4.3’
             char1 char2 char3 int1 int2 int3    double1    double2 double3
   1: id0000009348 id078   id1 3313   63    2 0.15461546 0.49494949    0.00
   2: id0000003862 id043   id4 8495    1    2 0.90929093 0.44444444    0.75
   3: id0000006380 id098   id4 9588   77    1 0.02410241 0.29292929    0.00
   4: id0000007731 id089   id3 6355   44    2 0.29822982 0.85858586    0.50
   5: id0000002179 id001   id3 5293   70    4 0.90779078 0.32323232    0.00
  ---                                                                      
1866: id0000006149 id041   id3 1573   65    3 0.45184518 0.82828283    0.25
1867: id0000000520 id080   id2 2914   16    5 0.24002400 0.09090909    0.00
1868: id0000005978 id077   id4 2264   92    1 0.21452145 0.84848485    1.00
1869: id0000008768 id044   id3 7501   25    2 0.62416242 0.69696970    1.00
1870: id0000002430 id033   id3 4688   42    2 0.44774477 0.79797980    0.75
```
output on latest 1.10.5
```
[1] ‘1.10.5’
Error: cannot allocate vector of size 6.5 Gb
```
@MarkusBonsch it might eventually be related to your subset PR.

### PR DESCRIPTION
**Pull request by jangorecki**
closes #2635

### CODE REVIEW

File: vignettes/datatable-benchmarking.Rmd [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+## fread: clear caches
+
+Ideally each `fread` call should be run in fresh session with the following commands preceding R execution. This clears OS cache file in RAM and HD cache.  
+
+```sh
+free -g
+sudo sh -c 'echo 3 >/proc/sys/vm/drop_caches'
+sudo lshw -class disk
+sudo hdparm -t /dev/sda
```
Comment (MichaelChirico) [inline vignettes/datatable-benchmarking.Rmd]: could go for some explication on what these commands are doing

Comment (jangorecki) [reply vignettes/datatable-benchmarking.Rmd]: General description is provided above the code chunk. User can always access manual of those unix commands.

**Comment by MichaelChirico** [comment]
As it stands I think this needs to better motivated. Quite dense now. Perhaps include some examples of the biases created when things are done wrong.

**Comment by jangorecki** [comment]
@MichaelChirico you are welcome to push directly to this branch

**Comment by telenz** [comment]
Great idea. I have added some piece of information that I find important and I have updated the `datatable.optimize` man-page (I should have done that during implementation).

**Comment by MarkusBonsch** [comment]
Great idea, I have added a piece of information that I find important and I have updated the `datatable.optimize` man page accordingly (should have done that during implementation).

**Comment by jangorecki** [comment]
preview at https://rdatatable.gitlab.io/data.table/library/data.table/doc/datatable-benchmarking.html

**Pull request merged**